workbench-core/base/src/utilities/queryParamsFiltersUtils.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { marshall } from '@aws-sdk/util-dynamodb';
+import Boom from '@hapi/boom';
+import { QueryParams } from '../aws/helpers/dynamoDB/dynamoDBService';
+import { FilterRequest } from '../interfaces/FilterRequest';
+import { SortRequest } from '../interfaces/SortRequest';
+
+/************************************************************
+ * Validates only one sorting or filter in list requests
+ * @param filter - filter request object
+ * @param sort - sorting request object
+ ************************************************************/
+export function validateSingleSortAndFilter(filter?: FilterRequest, sort?: SortRequest): void {
+  const filterAttributesLength = filter ? Object.keys(filter).length : 0;
+  const sortAttributesLength = sort ? Object.keys(sort).length : 0;
+  // Check that at most one filter is defined because we not support more than one filter
+  if (filterAttributesLength > 1) {
+    throw Boom.badRequest('Cannot apply more than one filter.');
+  }
+  // Check that at most one sort attribute is defined because we not support sorting by more than one attribute
+  if (sortAttributesLength > 1) {
+    throw Boom.badRequest('Cannot sort by more than one attribute.');
+  }
+  const hasSameProperty =
+    filterAttributesLength === 1 &&
+    sortAttributesLength === 1 &&
+    Object.keys(filter ?? {})[0] === Object.keys(sort ?? {})[0];
+  if (filter && sort && !hasSameProperty) {
+    throw Boom.badRequest('Cannot apply a filter and sort at the same time');
```

Comment:
It looks like you can apply a filter and a stort at the same time as long as they're operating on the same property?

Reply:
yes, dynamo allows this , I tested it and works

Reply:
I think this error needs to be reworded then. Maybe something like `'Cannot apply a filter and sort to different properties at the same time'`

Reply:
validation text updated